core/src/main/clojure/xtdb/metadata.clj

```diff
@@ -376,68 +448,87 @@
 (deftype MetadataManager [^BufferAllocator allocator
                           ^BufferPool buffer-pool
                           ^Cache table-metadata-idx-cache
-                          ^NavigableMap blocks-metadata
-                          ^:volatile-mutable ^Map fields]
+                          ^:volatile-mutable ^Map last-block-metadata
+                          ^:volatile-mutable ^Map table->fields]
   IMetadataManager
+  ;; the new-block-metadata is only the delta for the new block
   (finishBlock [this block-idx new-block-metadata]
-    (.putObject buffer-pool (->block-metadata-obj-key block-idx) (write-block-metadata new-block-metadata))
-    (set! (.fields this) (merge-fields fields new-block-metadata))
-    (.put blocks-metadata block-idx new-block-metadata))
+    (when (or (nil? (:block-idx last-block-metadata)) (< ^long (:block-idx last-block-metadata) block-idx))
```

Comment:
not reviewing this in any great detail as I assume it's moving about in the next PR :slightly_smiling_face: